streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -1294,4 +1307,11 @@ private <VR, KO, VO> KTable<K, VR> doJoinOnForeignKey(final KTable<KO, VO> forei
             builder
         );
     }
+
+    private static void maybeSetOutputVersioned(final GraphNode tableNode,
```

Comment:
To avoid this helper, would it be worth to rewrite this class a little bit and ensure that `materializedInternal` is never `null`?

Reply:
Hm, this class is method used only for table methods for which it's valid for `materializedInternal` to be null, e.g., table filter, mapValues, etc, where the result is not necessarily materialized. I don't think there's a way to rewrite this code such that `materializedInternal` will never be null in these cases, because non-null `materializedInternal` means that the result is materialized, when that's not necessarily the case?